airbyte-cdk/python/airbyte_cdk/sources/streams/http/http.py

```diff
@@ -409,23 +409,22 @@ def read_records(
         stream_slice: Mapping[str, Any] = None,
         stream_state: Mapping[str, Any] = None,
     ) -> Iterable[Mapping[str, Any]]:
+        yield from self._read_pages(
```

Comment:
this seems to be returning more than what the method signature asks for right? is that intentional?

Reply:
`_read_pages` yields Mapping, which is one of the types in the StreamData union

Reply:
updated method signature